src/govuk/components/accordion/template.njk

```diff
@@ -4,22 +4,24 @@
 <div class="govuk-accordion {%- if params.classes %} {{ params.classes }}{% endif -%}" data-module="govuk-accordion" id="{{ id }}"
 {%- for attribute, value in params.attributes %} {{attribute}}="{{value}}"{% endfor %}>
   {% for item in params.items %}
-    <div class="govuk-accordion__section {% if item.expanded %}govuk-accordion__section--expanded{% endif %}">
-      <div class="govuk-accordion__section-header">
-        <h{{ headingLevel }} class="govuk-accordion__section-heading">
-          <span class="govuk-accordion__section-button" id="{{ id }}-heading-{{ loop.index }}">
-            {{ item.heading.html | safe if item.heading.html else item.heading.text }}
-          </span>
-        </h{{ headingLevel }}>
-        {% if item.summary.html or item.summary.text %}
-          <div class="govuk-accordion__section-summary govuk-body" id="{{ id }}-summary-{{ loop.index }}">
-            {{ item.summary.html | safe if item.summary.html else item.summary.text }}
-          </div>
-        {% endif %}
+    {% if item %}
+      <div class="govuk-accordion__section {% if item.expanded %}govuk-accordion__section--expanded{% endif %}">
```

Comment:
By indenting the code it makes it easier to read but introduces more whitespace to the output HTML.

Should we consider keeping the if statement at the same level as the for loop?

This frustrates me very much I wish we could have best of both worlds.

Reply:
Yeah, I've got a similar issue with zoomtext where there is a value vs [space]value[space] issue to look at, I'll re-look at them.

Reply:
Presumably we could also filter the source items for empty - and then skip the if statement entirely.

Reply:
Is there already a filter for this? I'd be interested in seeing some render performance tests!

Reply:
I don't think there is a filter for this unfortunately, but if there were this would be the way to go.

Reply:
I've done some stuff with the mighty dash to clean up this example

base is current
new is from this pr
refactored is with - to clean up the input

Files, results and some janky performance 
https://gist.github.com/htmlandbacon/26c9119d6f0609b7e420f6a9a072e01e

Reply:
What do you think about this:

To get this merged let's do the simplest approach that doesn't impact whitespace and then consider refactoring separately as most of these templates could use some love regardless of Ed's original issue.

Reply:
I guess the easiest thing would be to leave, as is, with bonus white pace but if its going to get blocked till I fix them I can do!

Reply:
I don't think it's a hard blocker since the Design System website *should* reformat the HTML nicely, but I'd prefer if we tried to keep the output HTML from getting more funky even if it's a bit rough around the edges right now.

Will see what the rest of the team thinks later :)